TempMail/ContentView.swift

```diff
@@ -8,49 +8,110 @@
 
 import SwiftUI
 
+
 struct ContentView: View {
+    @State var orientation: UIDeviceOrientation = UIDevice.current.orientation
 //    variable for Email struct
     @ObservedObject var email: Email = Email()
     //this variable is for later, when I implement a feature that will save user data even when the app is closed
     private let defaults = UserDefaults.standard
     
+    // TODO: can the header be abstracted? lots of repeated code...
     var body: some View {
-        VStack {
-//          title
-            Text("Temp Mail")
-                .font(.largeTitle)
-                .padding()
-            Spacer()
-//          Generate new email button
-            Button(action: {
-                self.email.setNewEmailAddr()
-            }) {
-                Text("Generate New Email")
-                    .fontWeight(.bold)
-                    .foregroundColor(.blue)
-                    .padding()
-                    .overlay(RoundedRectangle(cornerRadius: 40)
-                        .stroke(Color.blue, lineWidth: 5))
-            }
-            
-            Spacer()
-//          display email address for user
-            Text("Current Address: \(self.email.getEmailAddr())@1secmail.com")
-                .padding()
-            
-            Spacer()
-//          List of emails attached to above address
-            List {
-                MessageRow(messageInfo: InboxModel(id: 1234,
-                                                   from: "sender@example.com",
-                                                   subject: "example subject",
-                                                   date: "Today"))
-                MessageRow(messageInfo: InboxModel(id: 1234,
-                                                   from: "sender@example.com",
-                                                   subject: "example subject",
-                                                   date: "Today"))
+        return HStack {
+            if orientation.isPortrait || orientation.isFlat { // TODO: test more to ensure that this is desired behavior
+                VStack {
+                    Text("Temp Mail")
+                        .font(.largeTitle)
+                        .padding()
+                    Spacer()
+                    
+                    Button(action: {
+                        self.email.setNewEmailAddr()
+                    }) {
+                        Text("Generate New Email")
```

Comment:
I think the email address should be displayed above the generate new email button, as this would conform better to the low-fidelity prototype, and the design idea that the most important things should be shown first (which is at the top in this case, because the user reads top to bottom). What do you think?

Reply:
I think it look ugly when you put the address before the button
![IMG_8611](https://user-images.githubusercontent.com/3253092/83945156-c2b98700-a7d6-11ea-983e-6331bc281183.PNG)

Reply:
Hm you may be right, it's not a big fix anyway, we can finish everything and tweak the design after user testing